iree/compiler/Codegen/LLVMGPU/ConvertToLLVM.cpp

```diff
@@ -256,14 +265,72 @@ class ConvertIREEBindingOp : public ConvertToLLVMPattern {
           rewriter, loc, *getTypeConverter(), memrefType, llvmBufferBasePtr);
       rewriter.replaceOp(op, {desc});
     } else {
-      // TODO: pull those paramters from HAL constants.
-      assert(0 && "TODO: implement dynamic shape");
+      ValueRange dynamicDims = adaptor.dynamic_dims();
```

Comment:
It looks this duplicates the logic from CPU side. It would be nice to put at some common place. But I'm not familiar enough about the ABI difference (if there is any) between CUDA side and CPU to say for sure. :)

Reply:
The ABI for CUDA and CPU are quite different overall, however this piece of code is very similar. It could be shared in a common helper but I’m not sure it is worth the extra dependency?

Reply:
Oh, I was just under the impression that CUDA and CPU both uses MemRef descriptor so that's why my question. If it does not really make sense, feel free to ignore. :)